bin/hdfcoinc/pycbc_multiifo_add_statmap

```diff
@@ -245,14 +295,214 @@ if not injection_style:
+            f['foreground_h%s/%s' % (h_iterations, key)] = \
+                    combined_fg_data.data[key][:]
+
+    h_iterations += 1
+
+    max_each_combo = {combo: sep_fg_data[combo].data['ifar'][:].argmax()
+                      for combo in all_ifo_combos}
+    max_ifars = {combo: sep_fg_data[combo].data['ifar'][:][max_each_combo[combo]]
+                    for combo in all_ifo_combos}
+    logging.info('Maximum IFAR values per combination:')
+    for k in max_ifars:
+        logging.info('{}: {:.3g}'.format(k, max_ifars[k]))
+    maxcombo = max(max_ifars, key=lambda key: max_ifars[key])
+    max_ifar_idx = max_each_combo[maxcombo]
+    max_ifar = sep_fg_data[maxcombo].data['ifar'][:][max_ifar_idx]
+    if not max_ifar > args.hierarchical_removal_ifar_limit:
+        logging.info("Loudest Event IFAR of %.3f in %s is less than minimum of %f"
+                     % (max_ifar, maxcombo, args.hierarchical_removal_ifar_limit))
+        any_above_ifar_lim = False
+        break
+    #Add the highest ifar (yet to be removed) to the final output
+    final_fg_data[maxcombo].__add__(sep_fg_data[maxcombo].select([max_ifar_idx]))
+
+    maxtime = pycbc.events.mean_if_greater_than_zero(
+        [sep_fg_data[maxcombo].data['%s/time' % ifo][:][max_ifar_idx] for 
+         ifo in all_ifos if ifo in combo])[0]
+    logging.info('Removing trigger from {} foreground at time {:.2f} with '
+                 'ifar {:.3g}'.format(maxcombo, maxtime, max_ifar))
+    sep_fg_data[maxcombo] = sep_fg_data[maxcombo].remove(max_ifar_idx)
+    logging.info('Removing trigger from combined foreground')
+    where_combined = np.nonzero(combined_fg_data.data['stat'] == 
+                                sep_fg_data[maxcombo].data['stat'][:][max_ifar_idx])
+    combined_fg_data = combined_fg_data.remove(where_combined)
+    n_triggers -= 1
+    logging.info('Removing background triggers at time {} within window '
+                 '{}s'.format(maxtime,args.hierarchical_removal_window))
+    trigger_time = np.array([sep_fg_data[maxcombo].data['%s/time' % ifo][max_ifar_idx] for
+                          ifo in all_ifos if ifo in maxcombo]).mean()
+    start = [trigger_time - args.hierarchical_removal_window]
+    end = [trigger_time + args.hierarchical_removal_window]
+    trigger_ids = {ifo: sep_fg_data[maxcombo].data['%s/trigger_id' % ifo][max_ifar_idx] for
+                          ifo in all_ifos if ifo in maxcombo}
+    
+    for combo in all_ifo_combos:
+        all_hred_idx = []
+        for ifo in all_ifos:
+            if ifo in combo:
+                hred_ids = veto.indices_within_times(
+                              sep_bg_data[combo].data['%s/time' % ifo][:],
+                              start, end)
+                times = sep_bg_data[combo].data['%s/time' % ifo]
+            all_hred_idx += list(hred_ids)
+        sep_bg_data[combo] = sep_bg_data[combo].remove(all_hred_idx)
+
+    hred_ids = veto.indices_within_times(
```

Comment:
This part looks suspicious. Shouldn't this be a loop over all ifos?

Reply:
Yes it should - updated (also with slightly changed index finding method which is clearer than `indices_within_times`)